synapse/storage/data_stores/main/push_rule.py

```diff
@@ -77,6 +80,15 @@ class PushRulesWorkerStore(
     def __init__(self, database: Database, db_conn, hs):
         super(PushRulesWorkerStore, self).__init__(database, db_conn, hs)
 
+        if hs.config.worker.worker_app is None:
+            self._push_rules_stream_id_gen = ChainedIdGenerator(
+                self._stream_id_gen, db_conn, "push_rules_stream", "stream_id"
+            )  # type: Union[ChainedIdGenerator, SlavedIdTracker]
+        else:
+            self._push_rules_stream_id_gen = SlavedIdTracker(
+                db_conn, "push_rules_stream", "stream_id"
+            )
```

Comment:
it's not obvious to me why this code needs to move (at all, but particularly in the same PR as moving the events stream)?

Reply:
It needed to move either way (as we hadn't called `super().__init__` previously), and broadly it felt easier/better to move it here than to below the `super().__init__` call.